File: game-logger/Domain/Models/DTOs/GameStateDto.cs
using System;
using System.Collections.Generic;

namespace Domain.Models.DTOs
{
    public class GameStateDto
    {
        public World World { get; set; }

        public List<BotDto> Bots { get; set; }

        // Used for the bot state
        public Guid BotId { get; set; }

        public List<PopulationTier> PopulationTiers { get; set; }

        public static GameStateDto GetStaticFields(GameStateDto gameStateDto)
        {

            return new GameStateDto
            {
                World = World.GetStaticFields(gameStateDto.World),
                Bots = gameStateDto.Bots.ConvertAll(bot => BotDto.GetStaticFields(bot)),
                BotId = gameStateDto.BotId
            };
        }
        public static GameStateDto GetVariableFields(GameStateDto previousGSDto, GameStateDto currentGSDto)
        {
            var variableWorld = World.GetVariableFields(previousGSDto.World, currentGSDto.World);

            return new GameStateDto
            {
                World = variableWorld,
                Bots = currentGSDto.Bots.ConvertAll(bot => BotDto.GetVariableFields(
                  previousGSDto.Bots.Find(x => x.Id == bot.Id), bot)),
                BotId = currentGSDto.BotId
            };
        }


    }
}

File: starter-bots/JavaBot/src/main/java/Models/Dtos/GameStateDto.java
package Models.Dtos;

import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

import Models.PopulationTier;
import Models.World;

public class GameStateDto {
    private World world;
    List<BotDto> bots;
    UUID botID = null;
    List<PopulationTier> populationTiers;

    public GameStateDto() {
        bots = new ArrayList<BotDto>(); 
    }


    public List<PopulationTier> getPopulationTiers() {
        return populationTiers;
    }

    public void setPopulationTiers(List<PopulationTier> populationTiers) {
        this.populationTiers = populationTiers;
    }

    public GameStateDto(World world, UUID botID, List<BotDto> bots, List<PopulationTier> populationTiers) {
        this.world = world;
        this.botID = botID;
        this.bots = bots;
        this.populationTiers = populationTiers;
    }

    public GameStateDto(World world, UUID botID, List<BotDto> bots) {
        this.world = world;
        this.botID = botID;
        this.bots = bots;
    }

    public BotDto getBotByID(UUID id) {
        for (BotDto bot : bots) {
            if (bot.id.equals(id)) {
                return bot;
            }
        }
        return null;
    }

    public World getWorld() {
        return world;
    }

    public void setWorld(World world) {
        this.world = world;
    }

    public UUID getBotID() {
        return botID;
    }

    public void setBotID(UUID botID) {
        this.botID = botID;
    }

    public List<BotDto> getBots() {
        return bots;
    }

    public void setBots(List<BotDto> bots) {
        this.bots = bots;
    }

    @Override
    public String toString() {
        return "GameStateDto [botID=" + botID + ", bots=" + bots + ", populationTiers=" + populationTiers + ", world="
                + world + "]";
    }

}